src/network/plugins/base/base-network-plugin.ts

```diff
@@ -47,9 +49,27 @@ export default class BaseNetworkPlugin extends Events {
         throw `Invalid string PeedID [${peerId}]: ${e.message}`;
       }
     }
+
+
     try {
       let peer: Libp2pPeer = await this.network.libp2p.peerStore.get(peerId)
         .catch(e => null)
+
+      if(peer && !peer.metadata.get("timestamp")){
```

Comment:
This condition is wrong. If the peer is in peerStore, but the timestamp is not on metadata, we should load peerInfo from peerRouting and make sure that the peer is still online.